Support try/catch on the happy (nothrow) path (#1474)
FluxML/Zygote.jl: docs/src/limitations.md

## Try-catch statements

> Exceptions containing try-catch statements can be differentiated if the catch block is not reached (no error are thrown).

Reviewer: ```suggestion
Code containting try-catch blocks can be differentiated as long as no exception is actually thrown.
```
